codice/ddf
# DDF-3121 Removed spaces in .config files as a work-around for a Felix bug in itests

#### What does this PR do?
This PR address a Felix bug discovered by @mackncheesiest: [FELIX-5690](https://issues.apache.org/jira/browse/FELIX-5690) ConfigurationHandler should support ignoring whitespace between property key/equal sign & equal sign/property value. The issue was encountered in a custom DDF distribution in which the itests were failing becuase a `.config` file that included spaces around the `=`s was modified using https://github.com/ops4j/org.ops4j.pax.exam2/blob/exam-reactor-4.11.0/containers/pax-exam-container-karaf/src/main/java/org/ops4j/pax/exam/karaf/options/KarafDistributionOption.java#L157-L160.

This PR also includes a commit to escape spaces in `.config` string values to match the guideline included in the Documentation.
#### Who is reviewing it? 
@mackncheesiest @figliold @vinamartin 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR.
@bdeining
@lessarderic
#### How should this be tested?
- Check that all default DDF `.config` file configurations still work and are able to be configured through the Admin Console.
- Add a deafult `.config` file to the distribution (with spaces around the `=`s), and modify it in the itests using https://github.com/ops4j/org.ops4j.pax.exam2/blob/exam-reactor-4.11.0/containers/pax-exam-container-karaf/src/main/java/org/ops4j/pax/exam/karaf/options/KarafDistributionOption.java#L157-L160. Confirm that the `.config` file is modified appropriately.
#### Any background context you want to provide?
This change does not affect DDF nor DDF itests because there are no `.config` files being modified in the itests. However, it protects against this issue in the future.
#### What are the relevant tickets?
[DDF-3121](https://codice.atlassian.net/browse/DDF-3121)
#### Screenshots (if appropriate)
![documentation-update](https://user-images.githubusercontent.com/8041246/30492975-5033e288-99f7-11e7-8dc6-e4464973ee3b.png)
#### Checklist:
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
